### ISSUE TITLE
include tasks in handlers fails if there are more handlers

### ISSUE THREAD

**Issue by kustodian** [ansible/ansible#34505]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
include

##### ANSIBLE VERSION
2.4.2 and devel
```
ansible 2.5.0 (devel 89670133c1) last updated 2017/12/25 12:55:15 (GMT +200)
```

##### SUMMARY
If we create a handler which uses the `include` directive and before it there is an a handler defined, the `include` handler doesn't work - Ansible cannot find it.

##### STEPS TO REPRODUCE
If we have a simple role `foo` and a playbook which runs this role:
```yaml
$ tree 
.
├── roles
│   └── foo
│       ├── handlers
│       │   ├── main.yml
│       │   └── test_handler.yml
│       └── tasks
│           └── main.yml
└── site.yml
```
```yml
# roles/foo/tasks/main.yml
---
- command: echo
  notify:
    - test handler
    - second handler
```
```yml
# roles/foo/handlers/main.yml
---
- name: second handler
  debug:
    msg: second handler

- name: test handler
  include: test_handler.yml
```
```yml
# roles/foo/handlers/test_handler.yml 
---
- debug:
    msg: first task

- debug:
    msg: second task

```
```yml
# site.yml
---
- hosts: localhost
  connection: local
  roles:
    - foo
```

##### EXPECTED RESULTS
Expected results is that both handlers are executed without any errors.

##### ACTUAL RESULTS
Ansible cannot find the `include` handler:

```
$ ansible-playbook site.yml 
PLAY [localhost] ***************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *********************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [foo : command] ***********************************************************************************************************************************************************************************************************************
ERROR! The requested handler 'test handler' was not found in either the main handlers list nor in the listening handlers list

```
If we swap the order in which the handlers are defined all works good:
```
$ ansible-playbook site.yml 
PLAY [localhost] ***************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *********************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [foo : command] ***********************************************************************************************************************************************************************************************************************
changed: [localhost]

RUNNING HANDLER [foo : debug] **************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "first task"
}

RUNNING HANDLER [foo : debug] **************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "second task"
}

RUNNING HANDLER [foo : second handler] *****************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "second handler"
}

PLAY RECAP *********************************************************************************************************************************************************************************************************************************
localhost                  : ok=5    changed=1    unreachable=0    failed=0 
```
or if we set the include as `static: no` like this:
```yml
- name: test handler
  include: handlers/test_handler.yml
  static: no
```
*NOTE:* For some reason I needed to add `handlers/` to the path to the file when I added `static: no`, not sure why is that.

Now if I run it again, all works:
```
$ ansible-playbook site.yml      
PLAY [localhost] ***************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *********************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [foo : command] ***********************************************************************************************************************************************************************************************************************
changed: [localhost]

RUNNING HANDLER [foo : second handler] *****************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "second handler"
}

RUNNING HANDLER [foo : test handler] *******************************************************************************************************************************************************************************************************
included: /home/kustodian/temp/handler_bug/roles/foo/handlers/test_handler.yml for localhost

RUNNING HANDLER [foo : debug] **************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "first task"
}

RUNNING HANDLER [foo : debug] **************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "second task"
}

PLAY RECAP *********************************************************************************************************************************************************************************************************************************
localhost                  : ok=6    changed=1    unreachable=0    failed=0
```

It is the exact same problem with `include_tasks`/`import_tasks`.

To sum it up:
 - `static: yes`/`import_tasks` works only if it's the first handler in the file
 - `static:  no`/`include_tasks` works always

**Comment by kustodian** [comment]
One interesting thing is that if you use a `import_tasks`/`static: yes` you can use this:
```yml
- name: test handler
  import_tasks: test_handler.yml
```
but if you use a dynamic include you have to add `handlers/`
```yml
- name: test handler
  include_tasks: handlers/test_handler.yml
```
Which feels like a different bug.

**Comment by alikins** [comment]
pinging @jimi-c

**Comment by beenje** [comment]
The difference between import_tasks / include_tasks search path might be linked to https://github.com/ansible/ansible/issues/33784

**Comment by sivel** [comment]
This issue was resolved in a889576d390eeab5137d31bd30821b3310c8d59e

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project
